charts/external-dns/templates/deployment.yaml

```
@@ -61,6 +61,9 @@ spec:
       {{- with .Values.dnsPolicy }}
       dnsPolicy: {{ . }}
       {{- end }}
+      {{- with .Values.dnsConfig }}
+      dnsConfig: {{ . }}
+      {{- end }}
```

```suggestion
      {{- with .Values.dnsConfig }}
      dnsConfig:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```
As `dnsConfig` is an object it needs to use this pattern.